sdk/search/azure-search/src/main/java/com/azure/search/SearchApiKeyPipelinePolicy.java

```diff
@@ -32,6 +32,9 @@ public SearchApiKeyPipelinePolicy(SearchApiKeyCredential apiKey) {
 
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        if ("http".equals(context.getHttpRequest().getUrl().getProtocol())) {
+            return Mono.error(new RuntimeException("token credentials require a URL using the HTTPS protocol scheme"));
```

Comment:
Is this using the same exception message as credentials in AAD use? It would be nice to turn it into a proper sentence here and file an issue for the other locations.

Reply:
Thanks for catching this.